source/SIL.AppBuilder.Portal.Frontend/src/ui/components/login-custom.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { withTemplateHelpers, Mut } from 'react-action-decorators';
+
+import { login as loginToAuth0 } from '@lib/auth0';
+
+export interface IProps {
+  afterLogin: () => void;
+}
+
+export interface IState {
+  isPasswordFieldVisible: boolean;
+  username?: string;
+  password?: string;
+}
+
+const EMAIL_REGEX = /(.+)@(.+)\.(.+)/;
```

Comment:
const emailPattern = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

This pattern is more complex and ISO complaint if I remember correctly

Reply:
yeah, I have strong opinions about email regex. we really don't need something that complicated, and that regex is garbage anayway. We don't need to perfectly validate domains, we just need the required parts of an email -- the before and after the `@` and the before and after the `.`.

Take a look at this: https://davidcel.is/posts/stop-validating-email-addresses-with-regex/
it takes the idea even further by only requiring an `@`.

The point is that we don't need to be over aggressive in telling people that they're wrong, and we have no idea how email addresses will look in the future. For example: a toooooooooooon of websites break on `+tags` in email addresses... but they are fairly common now to track spam / data selling. example: `email+somewebsite@gmail.com`.